crates/chain/src/spk_txout_index.rs

```diff
@@ -52,6 +52,35 @@ impl<I> Default for SpkTxOutIndex<I> {
     }
 }
 
+impl<I: Clone + Ord + 'static> TxIndex for SpkTxOutIndex<I> {
+    type Additions = BTreeSet<I>;
+
+    type SpkIndex = I;
+
+    fn index_txout(&mut self, outpoint: OutPoint, txout: &TxOut) -> Self::Additions {
+        self.scan_txout(outpoint, txout)
+            .cloned()
+            .into_iter()
+            .collect()
+    }
+
+    fn index_tx(&mut self, tx: &Transaction) -> Self::Additions {
+        self.scan(tx)
+    }
+
+    fn apply_additions(&mut self, _additions: Self::Additions) {
+        // This applies nothing.
```

Comment:
then why isn't the associated type for `Additions` just `()`.

Reply:
Done in a63ffe97397cd14bc0a13ea5e96ceddf8b63a4f0